File: TLBB1.5/tabletools/CodeTable/CSharp/Table_Worldbosspar.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_Worldbosspar : ITableOperate
{ private const string TAB_FILE_DATA = "worldbosspar.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_HP_PAR,
ID_ATT_PAR,
ID_GOLD_PAR,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private int m_AttPar;
 public int AttPar { get{ return m_AttPar;}}

private int m_GoldPar;
 public int GoldPar { get{ return m_GoldPar;}}

private int m_HpPar;
 public int HpPar { get{ return m_HpPar;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_Worldbosspar _values = new Tab_Worldbosspar();
 _values.m_AttPar =  Convert.ToInt32(valuesList[(int)_ID.ID_ATT_PAR] as string);
_values.m_GoldPar =  Convert.ToInt32(valuesList[(int)_ID.ID_GOLD_PAR] as string);
_values.m_HpPar =  Convert.ToInt32(valuesList[(int)_ID.ID_HP_PAR] as string);

 _hash.Add(nKey,_values); }


}
}

File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_Worldbosspar.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_Worldbosspar implements ITableOperate {
 public final static String TAB_FILE_DATA = "/worldbosspar.txt";
 public static final int ID_HP_PAR = 0 ;
public static final int ID_ATT_PAR = 1 ;
public static final int ID_GOLD_PAR = 2 ;
public static final int MAX_RECORD = 3 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private int m_AttPar;
 public int GetAttPar() { return m_AttPar;}

private int m_GoldPar;
 public int GetGoldPar() { return m_GoldPar;}

private int m_HpPar;
 public int GetHpPar() { return m_HpPar;}

public boolean LoadTable(HashMap<Integer, Table_Worldbosspar> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_Worldbosspar> _hash = (HashMap<Integer, Table_Worldbosspar> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_Worldbosspar _values = new Table_Worldbosspar();
 _values.m_AttPar =  Integer.parseInt((String)valuesList.get(ID_ATT_PAR));
_values.m_GoldPar =  Integer.parseInt((String)valuesList.get(ID_GOLD_PAR));
_values.m_HpPar =  Integer.parseInt((String)valuesList.get(ID_HP_PAR));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

